### DISCUSSION TITLE
Show/hide conditional field

### DISCUSSION THREAD

**Discussion by dominikkrulak** [craftcms/cms#13471] [Q&A]
I have two conditional fields `backgroundColor` and `backgroundGradientDirection` (both dropdown fields).
The background color field has as the title suggests background colors and background gradient colors which depend on background gradient direction.

So for example when I select _poppy_ gradient color the conditional field `backgroundGradientDirection` shows up to choose directions from. When both fields are saved all work as expected.

But than when I edit the background color field and change it to some plain color like _blue_ the field background gradient direction hides and than I save them the value from `backgroundGradientDirection` is intact and available on front end.

I'd like to ask if this is expected behavior.

**Comment by thupsi** [comment]
I think this is expected behavior. The conditions only affect the visibility of fields in the edit entry screen. Front end logic should be handled in your templates.

**Reply by dominikkrulak** [reply]
Yes that's what it says and I get it. But it isn't logical to me. I rather see the hidden value resetted back to default value (if set by default) or `null` or `""` empty string. Depends on data type. 

That would make much less noise on front end.

**Reply by thupsi** [reply]
Clearing the value when a field is hidden could introduce issues for authors when they switch a field on and off to try things out. I get what you're saying though and maybe there's an elegant way to solve this.
